returnn/datasets/concat_files.py

```diff
@@ -354,6 +355,11 @@ def _get_files_per_sub_epochs(
         assert all(files_per_sub_epochs)
         return files_per_sub_epochs
 
+    @staticmethod
+    def _get_key_for_file_tree(t: FileTree) -> str:
```

Comment:
I think you can move that to the global scope, instead of making a static method here. The whole file here (`concat_files.py`) is only about the dataset. That makes the code a bit shorter. Also, other things here are also moved out of the class.

Reply:
Hm, I'm not sure. I think the key format should remain specific to the dataset and be un-generic. I don't want other parts of the code to start relying on the concrete representation this function produces. If this comes up again we can always move it somewhere global.

Reply:
I mean also as a hidden/private function (`_get_key_for_file_tree`), i.e. that no other parts of the code will use this.